corehq/apps/app_execution/api.py

```diff
@@ -189,6 +191,15 @@ def __exit__(self, exc_type, exc_val, exc_tb):
     def clone(self):
         return dataclasses.replace(self, data=copy.deepcopy(self.data) if self.data else None)
 
+    @cached_property
+    def app_build_id(self):
+        app = get_app(self.client.domain, self.app_id, latest=True)
+        build_on = "Latest Version"
+        if app.built_on:
+            build_on = app.built_on.strftime("%B %d, %Y")
+        print(f"Using app '{app.name}' ({app._id} - {build_on})", file=self.log)
```

Comment:
Might be useful to log the app version number too - easier to cross reference on the releases page.